### ISSUE TITLE
[BUG]: Implicit use of 'self' in closure; use 'self.' to make capture semantics explicit

### ISSUE THREAD

**Issue by phamKhacQuang9x** [TheWidlarzGroup/react-native-video#3764]
### Version

6.0.0-rc.1

### What platforms are you having the problem on?

iOS

### Architecture

New architecture with interop layer

### What happened?

A bug happened!

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are:

**Comment by freeboub** [comment]
cannot reproduce on sample. Need more details

**Comment by waheedakhtar694** [comment]
@freeboub  I am getting the same error, I was using version `5.2.x` and after upgrading to version `6.0.0` i am getting this.
<img width="263" alt="Screenshot 2024-05-18 at 5 15 59 AM" src="https://github.com/TheWidlarzGroup/react-native-video/assets/45425486/4902088f-f860-4346-92fa-b77015f4abc3">

**Comment by dibyendu** [comment]
For anyone hitting this page with this issue, the following patch works until it's patched in the code.

```patch
diff --git a/node_modules/react-native-video/ios/Video/NowPlayingInfoCenterManager.swift b/node_modules/react-native-video/ios/Video/NowPlayingInfoCenterManager.swift
index 2307e10..d34559b 100644
--- a/node_modules/react-native-video/ios/Video/NowPlayingInfoCenterManager.swift
+++ b/node_modules/react-native-video/ios/Video/NowPlayingInfoCenterManager.swift
@@ -150,7 +150,7 @@ class NowPlayingInfoCenterManager {
             guard let self, let player = self.currentPlayer else {
                 return .commandFailed
             }
-            let newTime = player.currentTime() - CMTime(seconds: SEEK_INTERVAL_SECONDS, preferredTimescale: .max)
+            let newTime = player.currentTime() - CMTime(seconds: self.SEEK_INTERVAL_SECONDS, preferredTimescale: .max)
             player.seek(to: newTime)
             return .success
         }
@@ -160,7 +160,7 @@ class NowPlayingInfoCenterManager {
                 return .commandFailed
             }
 
-            let newTime = player.currentTime() + CMTime(seconds: SEEK_INTERVAL_SECONDS, preferredTimescale: .max)
+            let newTime = player.currentTime() + CMTime(seconds: self.SEEK_INTERVAL_SECONDS, preferredTimescale: .max)
             player.seek(to: newTime)
             return .success
         }
@@ -266,15 +266,15 @@ class NowPlayingInfoCenterManager {
 
             // case where there is new player that is not paused
             // In this case event is triggered by non currentPlayer
-            if rate != 0 && currentPlayer != player {
-                setCurrentPlayer(player: player)
+            if rate != 0 && self.currentPlayer != player {
+                self.setCurrentPlayer(player: player)
                 return
             }
 
             // case where currentPlayer was paused
             // In this case event is triggeret by currentPlayer
-            if rate == 0 && currentPlayer == player {
-                findNewCurrentPlayer()
+            if rate == 0 && self.currentPlayer == player {
+                self.findNewCurrentPlayer()
             }
         }
     }
diff --git a/node_modules/react-native-video/ios/Video/RCTVideo.swift b/node_modules/react-native-video/ios/Video/RCTVideo.swift
index 486524d..5b07d6b 100644
--- a/node_modules/react-native-video/ios/Video/RCTVideo.swift
+++ b/node_modules/react-native-video/ios/Video/RCTVideo.swift
@@ -523,7 +523,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
                     guard let self else { throw NSError(domain: "", code: 0, userInfo: nil) }
 
                     let playerItem = try await self.preparePlayerItem()
-                    try await setupPlayer(playerItem: playerItem)
+                    try await self.setupPlayer(playerItem: playerItem)
                 } catch {
                     DebugLog("An error occurred: \(error.localizedDescription)")
 
@@ -761,7 +761,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
             guard let self else { return }
 
             self._playerObserver.addTimeObserverIfNotSet()
-            self.setPaused(_paused)
+            self.setPaused(self._paused)
             self.onVideoSeek?(["currentTime": NSNumber(value: Float(CMTimeGetSeconds(item.currentTime()))),
                                "seekTime": seekTime,
                                "target": self.reactTag])
diff --git a/node_modules/react-native-video/ios/Video/RCTVideoManager.swift b/node_modules/react-native-video/ios/Video/RCTVideoManager.swift
index d8053f3..ff8e6ed 100644
--- a/node_modules/react-native-video/ios/Video/RCTVideoManager.swift
+++ b/node_modules/react-native-video/ios/Video/RCTVideoManager.swift
@@ -19,7 +19,7 @@ class RCTVideoManager: RCTViewManager {
             }
 
             guard let view = self.bridge.uiManager.view(forReactTag: reactTag) as? RCTVideo else {
-                RCTLogError("Invalid view returned from registry, expecting RCTVideo, got: \(String(describing: view))")
+                RCTLogError("Invalid view returned from registry, expecting RCTVideo, got: \(String(describing: self.view))")
                 callback(nil)
                 return
             }
```

**Comment by freeboub** [comment]
@dibyendu Thank you for the patch ! Can you please open a pull request with this patch please ? I don't see any issue with that patch. Thank you !

**Comment by dibyendu** [comment]
> @dibyendu Thank you for the patch ! Can you please open a pull request with this patch please ? I don't see any issue with that patch. Thank you !

Done.

#3881

**Comment by freeboub** [comment]
merged and released in 6.2.0 !
I just hope no other change did the same issue ...
Thank you for the pull request !
